bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/ide/IDEWorkbenchErrorHandler.java

```diff
@@ -133,9 +133,13 @@ private void handleException(Throwable t) {
 		try {
 			exceptionCount++;
 			if (exceptionCount > 1) {
-				dialog.updateMessage(MessageFormat.format(MSG_FATAL_ERROR,
-						MSG_FATAL_ERROR_Recursive));
-				dialog.getShell().forceActive();
+				try {
+					dialog.updateMessage(MessageFormat.format(MSG_FATAL_ERROR, MSG_FATAL_ERROR_Recursive));
+					dialog.getShell().forceActive();
+				} catch (Exception e) {
+					// avoid further explosion of recursive exceptions
+					e.printStackTrace();
```

Comment:
Please log this error (with appropriate message) not to console but to the log

Reply:
Why? the logToListeners would intercept the log and could run into this (or an other error) again. The mission is to find an end in the errorhandling. The first error that has initialted the trouble was already logged.